back-end/sports_platform/sports_platform_api/models/user_models.py

```diff
@@ -377,6 +379,108 @@ def get_spectating_events(self):
         except Exception as e:
             return 500
 
+    def get_badges(self):
+        
+        data = dict()
+
+        try:
+            user_given_badges = self.badges_by_users.all()
+
+            user_given_badges_list = []
+
+            for badge in user_given_badges:
+                item = dict()
+                if badge.badge.wikidata:
+                    item['@context'] = "https://www.wikidata.org/entity/" + badge.badge.wikidata
+                    item['name'] = badge.badge.name
+                    item['additionalProperty'] = {
+                        "@type": "PropertyValue",
+                        "name": "givenBy",
+                        "value": {
+                            "@context": "https://schema.org/Person",
+                            "@id": badge.from_user.user_id
+                        }
+                    }
+                else:
+                    item['name'] = badge.badge.name
+                    item['additionalProperty'] = {
+                        "@type": "PropertyValue",
+                        "name": "givenBy",
+                        "value": {
+                            "@context": "https://schema.org/Person",
+                            "@id": badge.user.user_id
```

Comment:
why here do we use `badge.user.user_id` instead of `badge.from.user_id`?

Reply:
Thanks for pointing out, fixed.